### ISSUE TITLE
Undefined index: eof

### ISSUE THREAD

**Issue by max-kut** [php-mime-mail-parser/php-mime-mail-parser#314]
When i set stream with `$parser->setStream($stream)`, an error occurs with a non-existent eof key
```
{
    "message": "Undefined index: eof",
    "exception": "ErrorException",
    "file": "/var/www/vendor/php-mime-mail-parser/php-mime-mail-parser/src/Parser.php",
    "line": 143
}
```
[Based on the documentation](https://www.php.net/manual/en/function.stream-get-meta-data.php), it is better to use the `feof()` to determine if the stream has reached the end.

Can I send PR?

**Comment by eXorus** [comment]
Yes of course you can send a PR.

**Comment by dsc-jose** [comment]
Hello,

Not sure if this is the correct place to ask but is this fix going to be merged to version 6? If not then I could fork this repo right?

Thanks

**Comment by eXorus** [comment]
Hello,

It will be fixed only in the next 7.0. 
Sorry for that I don't have enough time, and already have a lot of work to finish the next big version.

**Comment by dsc-jose** [comment]
No problem @eXorus , still a great library. I'll just fork it at the meantime and just switch to php7.3 in the future I guess. 

Thanks
